proto/spacemesh/v1/app_types.proto

```diff
@@ -0,0 +1,124 @@
+syntax = "proto3";
+package spacemesh.v1;
+option go_package = "github.com/spacemeshos/api/release/go/spacemesh/v1";
+import "spacemesh/v1/types.proto";
+
+// Spacemesh app primitive variable types
+enum VariableType {
+    VARIABLE_TYPE_UNSPECIFIED = 0;
+    VARIABLE_TYPE_I8 = 1;
+    VARIABLE_TYPE_U8 = 2;
+    VARIABLE_TYPE_I16 = 3;
+    VARIABLE_TYPE_U16 = 4;
+    VARIABLE_TYPE_I32 = 5;
+    VARIABLE_TYPE_U32 = 6;
+    VARIABLE_TYPE_I64 = 7;
+    VARIABLE_TYPE_U64 = 8;
+    VARIABLE_TYPE_BOOL = 9;
+    VARIABLE_TYPE_AMOUNT = 10;
+    VARIABLE_TYPE_ADDRESS = 11;
+}
+
+// Variable metadata
+message VariableMetadata {
+    uint32 id = 1; // var index in app's state
+    string name = 2; // var name
+    VariableType type = 3; // type
+    bool is_array = 4; // when true, variable is an array of 0 or more primitive value types, otherwise, a single primitive type value.
+}
+
+// The meta-data and value of an app's variable
+message Variable {
+    VariableMetadata meta_data = 1;
+    repeated bytes data = 2; // One item - primitive var data. More than one item - an array of primitive data.
+}
+
+// The address of an app template in the templates global state store
+message AppTemplateAddress {
```

Comment:
In the SVM code there is no `AppTemplateXXX` - so I'd rename it to `TemplateAddress`

Reply:
done